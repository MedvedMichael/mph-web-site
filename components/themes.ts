const light = {
    bg: {
        primary: '#c8d8e3',
        secondary: '#ffffff',
        inset: '#e2e4e8',
        input: 'rgba(65,67,78,0.12)',
        nav: '#5785ba'
    },
    text: {
        primary: '#fbfbfc',
        secondary: '#2f3037',
        tertiary: '#525560',
        quarternary: '#9194a1',
        placeholder: 'rgba(82,85,96,0.5)',
        onPrimary: '#ffffff'
    },
    border: {
        nav: '1px solid #282828'
    }
    // ...
}

const dark = {
    bg: {
        primary: '#162333',
        secondary: '#111111',
        inset: '#111111',
        input: 'rgba(191,193,201,0.12)',
        nav: '#18304d'
    },
    text: {
        primary: '#fbfbfc',
        secondary: '#e3e4e8',
        tertiary: '#a9abb6',
        quarternary: '#6c6f7e',
        placeholder: 'rgba(145,148,161,0.5)',
        onPrimary: '#050505'
    },
    border: {
        nav: '1px solid #282828'
    }
    // ...
}

const defaultTheme = {
    fontSizes: [
        '14px', // 0
        '16px', // 1
        '18px', // 2
        '22px', // 3
        '26px', // 4
        '32px', // 5
        '40px'  // 6
    ],
    fontWeights: {
        body: 400,
        subheading: 500,
        link: 600,
        bold: 700,
        heading: 800,
    },
    lineHeights: {
        body: 1.5,
        heading: 1.3,
        code: 1.6,
    },
    // ...
};

export const lightTheme = { ...defaultTheme, ...light }
export const darkTheme = { ...defaultTheme, ...dark }