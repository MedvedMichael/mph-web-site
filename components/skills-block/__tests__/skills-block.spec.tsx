import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import SkillsBlock, {SkillsBlockContainer, TechCard} from '../skills-block'
import * as React from 'react';
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import { lightTheme } from '../../themes'
import 'jest-styled-components'

test ('should have correct styles', () => {
    const res = renderer.create(<ThemeProvider theme={lightTheme}><SkillsBlockContainer/></ThemeProvider>).toJSON()
    expect(res).toHaveStyleRule('background', lightTheme.bg.inset)
    expect(res).toHaveStyleRule('transition', lightTheme.transition.bg)
    expect(res).toHaveStyleRule('color', lightTheme.text.primary)
})

test ('should render', () => {
    const res = shallow(<SkillsBlock/>)
    expect(toJson(res)).toMatchSnapshot()
})

test ('should render tech card', () => {
    const res = shallow(<TechCard path="path" title="title"/>)
    expect(toJson(res)).toMatchSnapshot()
})
const mockSetState = jest.fn();

jest.mock('react', () => ({
    ...jest.requireActual('react'),
    useState: initial => [initial, mockSetState]
}));

test ('should invert on click', () => {
    
    const res = shallow(<TechCard path="path" title="title"/>)
    
    res.dive().simulate('click')
    expect(mockSetState).toBeCalledTimes(1)
    expect(res.dive().get(0).props.style).toStrictEqual({})
})



// jest.mock('react', () => ({
//     ...jest.requireActual('react')
// }));