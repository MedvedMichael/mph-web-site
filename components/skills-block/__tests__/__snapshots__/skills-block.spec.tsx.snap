// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<styled.div>
  <styled.div>
    <VisibilityContainer
      from={
        Object {
          "opacity": 0,
          "transform": "translate3d(-1rem, 0, 0)",
        }
      }
      to={
        Object {
          "opacity": 1,
          "transform": "translate3d(0, 0, 0)",
        }
      }
    >
      <h3>
        I'm still studying at the best faculty in KPI, Kiev (my own opinionðŸ˜‰)
      </h3>
    </VisibilityContainer>
    <VisibilityContainer
      from={
        Object {
          "opacity": 0,
          "transform": "translate3d(-1rem, 0, 0)",
        }
      }
      to={
        Object {
          "opacity": 1,
          "transform": "translate3d(0, 0, 0)",
        }
      }
    >
      <h1>
        BUT!
      </h1>
    </VisibilityContainer>
    <VisibilityContainer
      from={
        Object {
          "opacity": 0,
          "transform": "translate3d(-1rem, 0, 0)",
        }
      }
      to={
        Object {
          "opacity": 1,
          "transform": "translate3d(0, 0, 0)",
        }
      }
    >
      <h3>
        I'm already a skilled developer!ðŸ˜Ž
      </h3>
    </VisibilityContainer>
  </styled.div>
  <div>
    <styled.div>
      <TechCard
        path="https://upload.wikimedia.org/wikipedia/commons/6/6a/JavaScript-logo.png"
        title="JavaScript"
      />
      <TechCard
        path="https://d2eip9sf3oo6c2.cloudfront.net/tags/images/000/000/026/full/react.png"
        title="React"
      />
      <TechCard
        path="https://upload.wikimedia.org/wikipedia/commons/thumb/6/61/HTML5_logo_and_wordmark.svg/1200px-HTML5_logo_and_wordmark.svg.png"
        title="HTML5"
      />
      <TechCard
        path="https://cdn.worldvectorlogo.com/logos/css3.svg"
        title="CSS3"
      />
      <TechCard
        path="https://www.appnovation.com/sites/default/files/2019-06/techservicelogo_NodeJS.svg"
        title="NodeJS"
      />
      <TechCard
        path="https://digital.ai/sites/default/files/pictures/styles/maxwidth_300/public/pt_logos/mongodb.png?itok=T7Bcj44-"
        title="MongoDB"
      />
      <TechCard
        path="https://cdn.auth0.com/blog/logos/nextjs-logo.png"
        title="Next.js"
      />
      <TechCard
        path="https://cdn.iconscout.com/icon/free/png-512/redux-283024.png"
        title="Redux"
      />
      <TechCard
        path="https://www.styled-components.com/atom.png"
        title="Styled components"
      />
      <TechCard
        path="https://upload.wikimedia.org/wikipedia/commons/thumb/4/4c/Typescript_logo_2020.svg/512px-Typescript_logo_2020.svg.png"
        title="TypeScript"
      />
      <TechCard
        path="https://www.drupal.org/files/issues/2019-12-27/heroku_logo.png"
        title="Heroku Deployment"
      />
      <TechCard
        path="https://installprogram.ru/wp-content/uploads/2018/04/TelegramLogo-1.png"
        title="Telegram Bot API"
      />
      <TechCard
        path="https://ps.w.org/custom-wp-rest-api/assets/icon-128x128.png?rev=1975404"
        title="REST API"
      />
      <TechCard
        path="https://cdn.auth0.com/blog/new-bootstrap4/logo.png"
        title="Bootstrap"
      />
      <TechCard
        path="https://diyhacking.com/wp-content/uploads/2017/04/Linux-Shell.png"
        title="Linux Terminal"
      />
    </styled.div>
  </div>
  <styled.div>
    <VisibilityContainer
      from={
        Object {
          "opacity": 0,
          "transform": "translate3d(-2rem, 0, 0)",
        }
      }
      to={
        Object {
          "opacity": 1,
          "transform": "translate3d(0, 0, 0)",
        }
      }
    >
      <h3>
        and more...
      </h3>
    </VisibilityContainer>
  </styled.div>
</styled.div>
`;

exports[`should render tech card 1`] = `
<styled.div
  onClick={[Function]}
  style={Object {}}
>
  <VisibilityContainer
    from={
      Object {
        "opacity": 0,
        "transform": "translate3d(4rem, 0, 0)",
      }
    }
    to={
      Object {
        "opacity": 1,
        "transform": "translate3d(0, 0, 0)",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <styled.img
        src="path"
      />
      <styled.span>
        title
      </styled.span>
    </div>
  </VisibilityContainer>
</styled.div>
`;
